Sync SURF prodcution updates (#110)
basvandervlies/cf_surfsara_lib: Changelog.md

* [Version: 0.9.0 (2018-08-24)](#version-090-2018-08-24)

<!-- vim-markdown-toc -->
# Version: 1.5.0 (2022-??-??)

SCL enhancements:

>  * drop support fro cfengine 3.7 and 3.10 code is removed

Reviewer: ```suggestion
 * drop support for CFEngine 3.7 and 3.10 Code is removed
```

 * Added `run_class` option to `copy_files`
 * `scl_mustach_copy` can now handle debug output for string and list
 * `scl_service_rotate_files` logic rewrite so that it works as expected
 * `scl_copy_files` and `scl_service_install_tarballs` fix for `run_bundle` did not work must use class `<file>_repaired`
 * added new scl library bodies:
     * `scl_cmd_kept`:  Only set class `{value}_succeded` when command exits with value `0`
 * The json filenames that must be loaded can now contain variable names, eg:
```
"ssh": {
   "json_files": [
                "soil_$(def.cluster_role).json"
   ]
```

>  *  New method for copy/expanding mustache templates `scl_mustache_service_autorun`, Each bundle can define tenmplates to be used, eg:

Reviewer: ```suggestion
 *  New method for copy/expanding mustache templates `scl_mustache_service_autorun`, Each bundle can define templates to be used, e.g.:
```

>  *  New method for copy/expanding mustache templates `scl_mustache_service_autorun`, Each bundle can define tenmplates to be used, eg:

Reviewer: ```suggestion
 *  New method for copy/expanding mustache templates `scl_mustache_service_autorun`, Each bundle can define templates to be used, e.g.:
```

 * ssh
    * remove debian_8 setup
    * Added `scl_service_copy_dirs` functionality
    * added new class `SSH_HOST_CERTIFICATE`
      * will generate `$(ssh.config_dir)/ssh_known_hosts2` file with the aid of json variable `scl.ssh.cert_authorities`
      * [ssh host certificate setup]( https://berndbausch.medium.com/ssh-certificates-a45bdcdfac39)
```
"ssh": {
   "classes": }
        "HOST_CERTIFICATE": "any"
   },
   "cert_authorities": [
        {
        "servers": "*",
        "key": "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIA9mwksJWzluDF8ZungY2TiOTYVb6TmuTNi8AsG5+hJa",
        "comment": "CA-host@clustercomputing"
        }
    ]
```
 * slurm:
    * Added slurm major version as class based on `current_version` definition, eg: `SLURM_21_08`
    * When class `SLURM_CONFIGLESS_CONF_LINKS` is unset, remove the create symbolic links
    * use `mode_file` and `mode_dir` to be compatible with cfengine 3,20 and higher
    * SLURM version 22.05 and higher do not need file `cgroup_allowed_devices_file.conf`
    * `interactive_step_options` added as json parameter. How start interactive job via salloc

>     * `launch_parameters` set  launch parameters for job launch plugin, default: `use_interactive_step`

Reviewer: ```suggestion
    * `launch_parameters` set launch parameters for job launch plugin, default: `use_interactive_step`
```

>     * `launch_parameters` set  launch parameters for job launch plugin, default: `use_interactive_step`

Reviewer: ```suggestion
    * `launch_parameters` set launch parameters for job launch plugin, default: `use_interactive_step`
```

>     * `launch_parameters` set  launch parameters for job launch plugin, default: `use_interactive_step`

Reviewer: ```suggestion
    * `launch_parameters` set launch parameters for job launch plugin, default: `use_interactive_step`
```
